ckanext/data_qld/de_identified_data/helpers.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from ckan import authz
+
+log = logging.getLogger(__name__)
+
+
+def process_de_identified_data_dict(data_dict, user_obj):
+    """
+    Remove de_identified_data value from dict
+    if current user doesn't have access to it.
+    """
+    # Add default value to NO.
+    if data_dict.get('de_identified_data', None) is None:
+        data_dict['de_identified_data'] = 'NO'
+
+    if not show_de_identified_data(data_dict, user_obj):
+        del data_dict['de_identified_data']
+
+
+def user_has_adm_editor_org_access(group_id, user_id):
+    """
+    Return False if the user does not have
+    admin, editor, or sysadmin access to the datasets organisation.
+    """
+    role = authz.users_role_for_group_or_org(group_id, user_id)
+    if role == 'admin' or role == 'editor':
+        return True
+
+    return False
+
+
+def show_de_identified_data(data_dict, user_obj):
+    """
+    Return False if current user doesn't have access to it.
+    """
+    is_sysadmin = False
+    has_org_admin_editor_access = False
+    if user_obj is not None:
+        is_sysadmin = user_obj.sysadmin
+        has_org_admin_editor_access = user_has_adm_editor_org_access(data_dict.get('owner_org', None), user_obj.id)
```

Comment:
There is no need to call the method 'user_has_adm_editor_org_access' if user is sysadmin.
Refactor the code to check if user is sysadmin first and exit if true otherwise fallback to check for admin/editor org access

Reply:
refactored